### ISSUE TITLE
Improve messaging when trying to connect to an app in release mode

### ISSUE THREAD

**Issue by johnpryan** [Dart-Code/Dart-Code#4273]
When an app is running in release mode, and I run the "Open DevTools  in Web Browser" command, I get this warning:

![Screenshot 2022-11-17 at 12 31 48 PM](https://user-images.githubusercontent.com/1145719/202555628-eb6125c4-07dd-49ae-be6d-814bd69f6d6c.png)

Could we detect when the launch configuration's `flutterMode` is set to `release` and point users to the [documentation](https://docs.flutter.dev/perf/ui-performance#run-in-profile-mode)?

**Comment by DanTup** [comment]
@johnpryan are you using current stable Flutter? I changed this message in https://github.com/Dart-Code/Dart-Code/commit/68cf41fbbae47d5e8ca5895778086e4c76afbaf5 but it requires https://github.com/flutter/flutter/pull/106337 (in theory we might be able to check the mode another way, but since this is already I'm inclined to just wait for it to ship).

I will add a "More info" button that links to those docs though.

**Comment by johnpryan** [comment]
That's a good point, the [Flutter performance profiling](https://docs.flutter.dev/perf/ui-performance) page isn't quite right because it's also common to use DevTools in Debug mode. I'm not sure we have a place in our docs that says DevTools should only be run in Debug or Profile mode, not Release mode.

**Comment by DanTup** [comment]
Ah, in that case is there anything else to do here? The current message (when running on latest code) says:

> DevTools is not available for an app running in this mode.

We could make this more specific (release/profile mode), however it requires making some assumptions in VS Code that I'd rather avoid (there are many ways the user could start the app in profile/release mode that are harder for us to detect).

**Comment by johnpryan** [comment]
I would prefer if it  said "DevTools only supports Debug or Profile mode".  The current gives two possible explanations, which can lead to confusion and doesn't tell the user how to resolve the issue.

Is there a way to detect that the debug session isn't ready yet so that we can disambiguate these two scenarios?

**Comment by johnpryan** [comment]
> Isn't it the case that profile mode for Web doesn't support DevTools

That's a good point, maybe we could link to a compatibility table in the docs or something...

> In a future release, we could drop the "or does not support DevTools"

Sounds good

**Comment by johnpryan** [comment]
Related to https://github.com/flutter/devtools/issues/4050

**Comment by johnpryan** [comment]
I filed an issue to add a warning about web support https://github.com/flutter/website/issues/7887

**Comment by DanTup** [comment]
> That's a good point, maybe we could link to a compatibility table in the docs or something...

Is there an existing one I can link to? If not, if we can add one I can add a "more info" button here to link to it.

**Comment by johnpryan** [comment]
No, I don't think we have a compatibility table in the docs right now.

**Comment by DanTup** [comment]
@johnpryan is there still work to do here? We can add a link to the message if there's somewhere appropriate to link to. The text has been tweaked slightly since the issue was originally reported:

https://github.com/Dart-Code/Dart-Code/blob/5a6e97678052106379023fb7c80e9fed8bdd36c4/src/extension/commands/debug.ts#L139-L147

**Comment by johnpryan** [comment]
Here's the link to the docs on profile mode, which has a warning about web + profile mode: https://docs.flutter.dev/perf/ui-performance#run-in-profile-mode. It would be

**Comment by johnpryan** [comment]
"DevTools is not available for [mode] mode on device [deviceId]"  would definitely help here, the table at https://dart.dev/tools/dart-devtools is also handy, but doesn't contain any information about whether the user is in run/profile/release mode

**Comment by DanTup** [comment]
Ok, I've updated so it'll look like this:

![Screenshot 2023-03-30 at 14 11 36](https://user-images.githubusercontent.com/1078012/228846749-9f17b2a6-e91a-4d4d-81b8-054bfbb523a6.png)

It requires both https://github.com/flutter/flutter/pull/121239 and a change here in the VS Code extension, so it'll still show the old message until you're using updated versions of each.

I haven't added a link to the page above for now, since it doesn't mention modes - however if it's reasonable for that page to be updated to mention modes, we could easily add a button here.
